contracts/stake/src/wasm.rs

```diff
@@ -44,49 +45,51 @@ impl StakeContract {
         let counter = stake.counter();
 
         stake.increment_counter();
-        stake.insert_amount(value, dusk_abi::block_height());
+        stake.insert_amount(value, rusk_abi::block_height());
 
         // required since we're holding a mutable reference to a stake and
         // `dusk_abi::transact_raw` requires a mutable reference to the state
         drop(stake);
 
         // verify the signature is over the correct message
         let message = Self::stake_sign_message(counter, value);
-        let valid_signature = rusk_abi::verify_bls_sign(signature, pk, message);
+        let valid_signature =
+            rusk_abi::verify_bls(message, APK::from(&pk), signature);
 
         if !valid_signature {
             panic!("Invalid signature!");
         }
 
         // make call to transfer contract to transfer balance from the user to
         // this contract
-        let transfer = rusk_abi::transfer_contract();
-        let transaction = {
-            let address = rusk_abi::stake_contract();
-            let call =
-                Call::send_to_contract_transparent(address, value, proof);
-            call.to_transaction()
-        };
-
-        dusk_abi::transact_raw(self, &transfer, &transaction, 0)
+        let transfer = rusk_abi::transfer_module();
+        let transaction = RawTransaction::new(
+            "stake",
+            Stct2 {
+                address: rusk_abi::stake_module(),
+                value,
+                proof,
+            },
+        );
+
+        self.transact_raw(transfer, transaction)
```

Comment:
This will not work, since the transfer contract does not export a `stake` function. The function you want to call here is `stct`. I also don't particularly get why we're using `transact_raw` here, since it would be easier and more readable to just use `transact`.
Also I don't understand the introduction of `Stct2` here. The `Stct` struct is already present in `transfer-types` and is exactly the same.